### PR TITLE
[SPARK-38986][SQL] Prepend error class tag to error messages

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to prepend the error class tag to error messages. The `SparkThrowableHelper.getMessage()` method was modified for this purpose:
```scala
def getMessage(errorClass: String, messageParameters: Array[String]): String = {
  ...
    "[" + errorClass + "] " + String.format ...
  }
```

Other changes are related to tests where the expected error messages were extended by error class tags. Also I introduced new base trait `QueryErrorsSuiteBase` for test suites w/ two helper methods:
```scala
  def checkErrorClass(
      exception: Exception with SparkThrowable,
      errorClass: String,
      msg: String,
      sqlState: Option[String] = None,
      matchMsg: Boolean = false): Unit

  def validateParsingError(
      sqlText: String,
      errorClass: String,
      sqlState: String,
      message: String): Unit
```
The methods are supposed to be used in `QueryParsingErrorsSuite`, `QueryCompilationErrorsSuite`, `QueryExecutionErrorsSuite` and so on, to check error classes, messages, and sqlStates.

### Why are the changes needed?
The error class tags will be used as unique identifiers to google issues, and refer to Spark docs.

### Does this PR introduce _any_ user-facing change?
Yes, the PR changes user-facing error messages.

### How was this patch tested?
By running the modified test suites:
```
$ build/sbt "sql/testOnly *QueryExecutionErrorsSuite"
$ build/sbt "test:testOnly *QueryCompilationErrorsSuite"
$ build/sbt "test:testOnly *QueryCompilationErrorsDSv2Suite"
$ build/sbt "testOnly *QueryParsingErrorsSuite"
$ build/sbt "sql/testOnly *SQLQueryTestSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@ivoson @yutoacts @leesf @cloud-fan @HyukjinKwon @gengliangwang @panbingkun Could you review this PR, please.

**Comment by MaxGekk** [comment]
Also cc @srielau @karenfeng @anchovYu @entong

**Comment by MaxGekk** [comment]
All tests have passed already: https://github.com/MaxGekk/spark/actions/runs/2202086239.

File: core/src/main/scala/org/apache/spark/ErrorInfo.scala [hunk 1]
```diff
@@ -58,7 +58,8 @@ private[spark] object SparkThrowableHelper {
   def getMessage(errorClass: String, messageParameters: Array[String]): String = {
     val errorInfo = errorClassToInfoMap.getOrElse(errorClass,
       throw new IllegalArgumentException(s"Cannot find error class '$errorClass'"))
-    String.format(errorInfo.messageFormat.replaceAll("<[a-zA-Z0-9_-]+>", "%s"),
+    "[" + errorClass + "] " + String.format(
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/ErrorInfo.scala:61]: maybe:

```suggestion
    s"[$errorClass] " + String.format(
```

**Review by HyukjinKwon — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/ErrorInfo.scala (hunk 1 above)
Comment (MaxGekk) [reply core/src/main/scala/org/apache/spark/ErrorInfo.scala:61]: Let's do this later. I guess we will touch the method one more time at least in the future.

**Comment by MaxGekk** [comment]
GA passed. Merging to master.
Thank you, @cloud-fan and @HyukjinKwon for review.

**Pull request closed without merging**